pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go

```diff
@@ -208,6 +209,35 @@ func (pl *nonCSILimits) EventsToRegister() []framework.ClusterEvent {
 	}
 }
 
+// PreFilter invoked at the prefilter extension point
+//
+// If the pod only use secrets and configmaps, we'll skip the Filter phase
+func (pl *nonCSILimits) PreFilter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	// If the new pod doesn't have any volume attached to it, then we'll skip the Filter phase
+	if len(pod.Spec.Volumes) == 0 {
+		return nil, framework.NewStatus(framework.Skip)
+	}
+
+	length := 0
+	for _, vol := range pod.Spec.Volumes {
+		if vol.ConfigMap != nil || vol.Secret != nil {
```

Comment:
Actually how about vol.EmptyDir or vol.HostPath?

I think they also can return fast.

Based on the body of Filter() method in *nonCSILimits, I think it's safe to skip if there is no volume with: pl.filter.FilterVolume(vol) || vol.PersistentVolumeClaim != nil || vol.Ephemeral != nil.

This should be slightly more exact and the checks here are still O(1).